addons/website_sale_picking/tests/test_ui.py

```diff
@@ -6,12 +6,48 @@
 
 @tagged('-at_install', 'post_install')
 class TestUi(HttpCase):
+    def setUp(self):
+        super(TestUi, self).setUp()
+        # create a Chair floor protection product
+        self.env['product.product'].create({
+            'name': 'Chair floor protection',
+            'type': 'consu',
+            'website_published': True,
+            'list_price': 1000,
+        })
+        # create a Customizable Desk product
+        self.env['product.product'].create({
+            'name': 'Customizable Desk',
+            'type': 'consu',
+            'website_published': True,
+            'list_price': 1000,
+        })
+        # create a Warranty product
+        self.env['product.product'].create({
+            'name': 'Warranty',
+            'type': 'service',
+            'website_published': True,
+            'list_price': 20,
+        })
+
+
     def test_onsite_payment_tour(self):
         # Make sure at least one onsite payment option exists.
         self.env['delivery.carrier'].create({
             'delivery_type': 'onsite',
+            'is_published': True,
             'website_published': True,
             'name': 'Example shipping On Site',
-            'product_id': self.env.ref('website_sale_picking.onsite_delivery_product').id
+            'product_id': self.env.ref('website_sale_picking.onsite_delivery_product').id,
         })
-        self.start_tour('/shop', 'onsite_payment_tour', login='admin')
+        self.env['payment.provider'].create({
+            'name': 'Test Payment Provider',
+            'company_id': self.env.company.id,
+            'custom_mode': 'onsite',
+            'is_published': True,
+            'code': 'custom',
+            'state': 'test',
+            'redirect_form_view_id': 778,
```

Comment:
This can't be hardcoded either
```suggestion
            'redirect_form_view_id': self.env.ref("xmlid").id,
```
I don't think it's necessary to create a new provider however.
There is one that exists. From the code it seems that it should be displayed in case the on site carrier is published see `_get_compatible_providers`